### PR TITLE
Fix genesis deposit balances in Solidity.

### PR DESCRIPTION
**Pull request by 86667**
Clone of https://github.com/Zilliqa/zq2/pull/1948

```
curl -X POST   -H "Content-Type: application/json"   --data '{"jsonrpc":"2.0","method":"eth_getBalance","params":["0x00000000005A494C4445504F53495450524F5859", "latest"],"id":1}' http://localhost:4201/
{"jsonrpc":"2.0","id":1,"result":"0x2116545850052128000000"}
```

### CODE REVIEW

File: zilliqa/src/contracts/deposit_v1.sol [hunk 1]
```diff
@@ -137,7 +137,7 @@ contract DepositInit is UUPSUpgradeable {
         uint256 _maximumStakers,
         uint64 _blocksPerEpoch,
         InitialStaker[] memory initialStakers
-    ) public initializer {
+    ) public payable initializer {
```
Comment (JamesHinshelwood) [inline zilliqa/src/contracts/deposit_v1.sol:140]: Don't we need to change v2 as well?

File: zilliqa/src/exec.rs [hunk 2]
```diff
@@ -454,13 +454,14 @@ impl State {
         &mut self,
         creation_bytecode: Vec<u8>,
         override_address: Option<Address>,
+        amount: Option<u128>,
```
Comment (JamesHinshelwood) [inline zilliqa/src/exec.rs:457]: nit: `amount: None` doesn't make sense. Callers should just pass `0` if they don't want to send anything.

**Review by JamesHinshelwood — APPROVED** [review]

File: zilliqa/src/contracts/deposit_v1.sol (hunk 1 above)
Comment (shawn-zil) [reply zilliqa/src/contracts/deposit_v1.sol:140]: v2 doesn't seem to have this function.

File: zilliqa/src/exec.rs (hunk 2 above)
Comment (shawn-zil) [reply zilliqa/src/exec.rs:457]: Small fix, I'll do that now.

File: zilliqa/src/contracts/deposit_v1.sol (hunk 1 above)
Comment (86667) [reply zilliqa/src/contracts/deposit_v1.sol:140]: v1 does the initialising, further versions inherit the previous version's storage and so do not need to initialise.

Comment (JamesHinshelwood) [reply zilliqa/src/contracts/deposit_v1.sol:140]: Makes sense, thanks for explaining.

**Review by JamesHinshelwood — APPROVED** [review]

**Review by JamesHinshelwood — APPROVED** [review]

**Pull request merged**